src/doc/trpl/advanced-linking.md

```diff
@@ -0,0 +1,175 @@
+you as the crate consumer know the actual name). To link against the appropriate
+library for your system, you can use the `#[link(...)]` attribute on an empty
+extern block:
+
+``` no_run
+extern crate foo;
+
+#[link(name = "foonative3"]
+extern {}
+# fn main() {}
+```
+
+# Link args
+
+There is one other way to tell rustc how to customize linking, and that is via
+the `link_args` attribute. This attribute is applied to `extern` blocks and
+specifies raw flags which need to get passed to the linker when producing an
+artifact. An example usage would be:
+
+``` no_run
+#![feature(link_args)]
+
+#[link_args = "-foo -bar -baz"]
+extern {}
+# fn main() {}
+```
+
+Note that this feature is currently hidden behind the `feature(link_args)` gate
+because this is not a sanctioned way of performing linking. Right now rustc
+shells out to the system linker, so it makes sense to provide extra command line
+arguments, but this will not always be the case. In the future rustc may use
+LLVM directly to link native libraries in which case `link_args` will have no
+meaning.
+
+It is highly recommended to *not* use this attribute, and rather use the more
+formal `#[link(...)]` attribute on `extern` blocks instead.
+
+# Static linking
+
+Static linking refers to the process of creating output that contain all
+required libraries and so don't need libraries installed on every system where
+you want to use your compiled project. Rust libraries are statically linked by
+default so you can use Rust-created binaries and libraries without installing
+the Rust runtime everywhere. By contrast, native libraries are dynamically
+linked by default, but sometimes it can be useful to change this.
+
+Linking is a very platform dependant topic - on some platforms, static linking
+may not be possible at all! This section assumes some basic familiarity with
+linking on your platform on choice.
+
+## Linux
+
+By default, all Rust programs on Linux will link to the system libc along with
+a number of other libraries Let's look at an example on a 64-bit linux machine
+with GCC and glibc (by far the most common libc on Linux):
+
+``` text
+$ cat example.rs
+fn main() {}
+$ rustc example.rs
+$ ldd example
+        linux-vdso.so.1 =>  (0x00007ffd565fd000)
+        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa81889c000)
+        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa81867e000)
+        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa818475000)
+        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fa81825f000)
+        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa817e9a000)
+        /lib64/ld-linux-x86-64.so.2 (0x00007fa818cf9000)
+        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa817b93000)
+```
+
+Dynamic linking on Linux can be undesirable if you wish to target older
+machines as applications compiled aginst newer versions glibc are not
+guaranteed to run against older versions.
```

Comment:
Ah, if it's somewhat different to what you took away from the sentence then it clearly needs rephrasing :)
The thrust is that I'd consider static linking an easier solution than finding/compiling an older libc and then trying to compile against it. It also helps in more exotic situations, like non-glibc distros or even distros without a dynamic libc.

Reading again, I see the mistake is in my sloppy wording - it should be 'linked against' rather than 'compiled against'.